Katello/katello
# Fixes #37988 - migrate container repo naming to use slashes

#### What are the changes introduced in this pull request?
Changes the default container naming scheme to use slashes instead of dashes.

#### Considerations taken when implementing this change?
We don't want the change to happen for existing repositories. To do this, we rely on the fact that the `::Katello::Repository` model holds the container repository name, which gets changed on updates as a before_validation hook. When updating repositories in the UI, that changes the root, not the repository. So, there is no way for the container repository to be update just by updating things on the UI.

However, the naming scheme will change for content view publishes and promotions, since the environment repository gets updated.

Is this behavior okay? Should new content views with old repositories continue to use the old naming default? In a way, when publishing/promoting, it's like creating new repositories. If we need to keep the old default, it may complicate the logic a bit.

Plus, we want users to use this new naming scheme because it is better and more readable. If users don't like the new scheme, it's easy to go back to the old default by adding a custom registry naming scheme.

#### What are the testing steps for this pull request?
1) Create some container repos
2) Publish those repos to some CVs in multiple environments
3) Add in my change
4) Check that existing repositories use the same old scheme, even if they get updated
5) Check that the existing CV version container repos have the same old naming scheme
6) Check that promoting or publishing the CVs updates the naming scheme
7) Check that new repositories use the new naming scheme.
